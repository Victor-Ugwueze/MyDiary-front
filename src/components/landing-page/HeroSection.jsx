// react libraries
import React from 'react';

const HeroSection = () => (
  <main>
    <section className="container" id="how-it-works">
        <div>
            <h2>How it works</h2>
        </div>
        <div className="row">
            <div className="col-md">
            <p>My Diary, is an online journal
              that helps you to pen down and manage your thoughts and feelings.
              with a reminder that will help you not to miss out on writing your journal.
            </p>
            </div>
        </div>
    </section>
  </main>
);

export default HeroSection;
