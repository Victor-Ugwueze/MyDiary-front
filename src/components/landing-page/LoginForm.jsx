// react libraries
import React from 'react';

// third-party libraries
import PropTypes from 'prop-types';

// components
import Loader from '../general/loader/Loader';

const LoginForm = ({
  show, onSubmit, onChange, submitted
}) => (
  <div className={`tab-pane ${show ? 'selected' : ''}`} id='login'>
    <form encType='application/x-www-form-urlencoded' onSubmit={onSubmit}>
      {submitted && <Loader />}
      <div className='form-row'>
        <div className='form-heading'>
          <p>Log in to your account, to acces your journals</p>
        </div>
        <div className='alert error-flash errors hide-error' id='email-error' />
        <div className='row'>
          <input
            type='text'
            className='form-input'
            onChange={onChange}
            name='email'
            autoComplete='username'
            placeholder='Email Address'
          />
        </div>
        <div className='row'>
          <input
            type='password'
            className='form-input'
            onChange={onChange}
            name='password'
            autoComplete='current-password'
            placeholder='Password'
          />
        </div>
        <div>
          <input type='submit' value='Log in' className='btn btn-primary form-input' />
        </div>
        <p className='login-small-signup'>
          or{' '}
          <span>
            <a href='#'>Sign up</a>
          </span>
        </p>
      </div>
    </form>
  </div>
);

LoginForm.propTypes = {
  show: PropTypes.bool,
  onSubmit: PropTypes.func,
  onChange: PropTypes.func,
  submitted: PropTypes.bool,
};

export default LoginForm;
