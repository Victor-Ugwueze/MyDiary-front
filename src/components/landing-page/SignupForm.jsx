// react libraries
import React from 'react';

// third-party libraries
import PropTypes from 'prop-types';

// components
import Loader from '../general/loader/Loader';

const SignupForm = ({
  show, onSubmit, onChange, submitted
}) => (
  <div className={`tab-pane ${show ? 'selected' : ''}`} id='signup'>
    <form encType='application/x-www-form-urlencoded' onSubmit={onSubmit}>
      {submitted && <Loader />}
      <div className='form-heading'>
        <h4>Sign up to My Diary</h4>
        <p>Enjoy the expereinces of having all your thouhgts in one place</p>
      </div>
      <div className='form-row'>
        <p className='errors alert error-flash hide-error' id='signup-error' />
        <div className='row'>
          <div className='col-2 form-col'>
            <input
              type='text'
              className='form-input'
              onChange={onChange}
              name='firstName'
              placeholder='FIrst Name'
            />
          </div>
          <div className='col-2'>
            <input
              type='text'
              className='form-input'
              onChange={onChange}
              name='lastName'
              placeholder='Last Name'
            />
          </div>
        </div>
        <div className='row'>
          <input
            type='text'
            className='form-input'
            onChange={onChange}
            name='email'
            placeholder='Email Address'
          />
        </div>
        <div className='row'>
          <div className='col-2 form-col'>
            <input
              type='password'
              className='form-input'
              name='password'
              onChange={onChange}
              autoComplete='current-password'
              placeholder='Password'
            />
          </div>
          <div className='col-2'>
            <input
              type='password'
              className='form-input'
              name='confirmPassword'
              onChange={onChange}
              autoComplete='confirm-password'
              placeholder='Confirm Password'
            />
          </div>
        </div>
        <div className='row' id='terms-of-use'>
          <input type='checkbox' />
          <label>
            Agree to our
            <a href='#'>Terms of use</a> and
            <a href='#'>Privacy policy</a>
          </label>
        </div>
        <div>
          <input type='submit' value='Sign up' className='btn btn-primary form-input' />
        </div>
      </div>
    </form>
  </div>
);

SignupForm.propTypes = {
  show: PropTypes.bool,
  onSubmit: PropTypes.func,
  onChange: PropTypes.func,
  submitted: PropTypes.bool,
};

export default SignupForm;
