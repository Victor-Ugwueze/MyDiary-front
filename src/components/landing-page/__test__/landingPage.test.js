// React Libraries
import React from 'react';

// third-party libraries
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';


// modules importation
import SignupForm from '../SignupForm';
import SignupPage from '../LandingPage';
import LoginForm from '../LoginForm';
import HeroSection from '../HeroSection';

configure({
  adapter: new Adapter(),
});

// SnapShot Test
describe('Renders the singup page', () => {
  it('should render the signup form', () => {
    const component = shallow(
      <SignupForm />
    );
    expect(component).toMatchSnapshot();
  });

  it('should render the login form', () => {
    const component = shallow(
      <LoginForm />
    );
    expect(component).toMatchSnapshot();
  });


  it('should render the hero section of landing page', () => {
    const component = shallow(
      <HeroSection />
    );
    expect(component).toMatchSnapshot();
  });

  it('should render the signup page', () => {
    const component = shallow(
      <SignupPage />
    );
    expect(component).toMatchSnapshot();
  });
});
