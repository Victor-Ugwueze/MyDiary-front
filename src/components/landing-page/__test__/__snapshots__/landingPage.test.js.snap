// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders the singup page should render the hero section of landing page 1`] = `
<main>
  <section
    className="container"
    id="how-it-works"
  >
    <div>
      <h2>
        How it works
      </h2>
    </div>
    <div
      className="row"
    >
      <div
        className="col-md"
      >
        <p>
          My Diary, is an online journal that helps you to pen down and manage your thoughts and feelings. with a reminder that will help you not to miss out on writing your journal.
        </p>
      </div>
    </div>
  </section>
</main>
`;

exports[`Renders the singup page should render the login form 1`] = `
<div
  className="tab-pane "
  id="login"
>
  <form
    encType="application/x-www-form-urlencoded"
  >
    <div
      className="form-row"
    >
      <div
        className="form-heading"
      >
        <p>
          Log in to your account, to acces your journals
        </p>
      </div>
      <div
        className="alert error-flash errors hide-error"
        id="email-error"
      />
      <div
        className="row"
      >
        <input
          autoComplete="username"
          className="form-input"
          name="email"
          placeholder="Email Address"
          type="text"
        />
      </div>
      <div
        className="row"
      >
        <input
          autoComplete="current-password"
          className="form-input"
          name="password"
          placeholder="Password"
          type="password"
        />
      </div>
      <div>
        <input
          className="btn btn-primary form-input"
          type="submit"
          value="Log in"
        />
      </div>
      <p
        className="login-small-signup"
      >
        or 
        <span>
          <a
            href="#"
          >
            Sign up
          </a>
        </span>
      </p>
    </div>
  </form>
</div>
`;

exports[`Renders the singup page should render the signup form 1`] = `
<div
  className="tab-pane "
  id="signup"
>
  <form
    encType="application/x-www-form-urlencoded"
  >
    <div
      className="form-heading"
    >
      <h4>
        Sign up to My Diary
      </h4>
      <p>
        Enjoy the expereinces of having all your thouhgts in one place
      </p>
    </div>
    <div
      className="form-row"
    >
      <div
        className="errors alert error-flash hide-error"
        id="signup-error"
      />
      <div
        className="row"
      >
        <div
          className="col-2 form-col"
        >
          <input
            className="form-input"
            name="firstName"
            placeholder="FIrst Name"
            type="text"
          />
        </div>
        <div
          className="col-2"
        >
          <input
            className="form-input"
            name="lastName"
            placeholder="Last Name"
            type="text"
          />
        </div>
      </div>
      <div
        className="row"
      >
        <input
          className="form-input"
          name="email"
          placeholder="Email Address"
          type="text"
        />
      </div>
      <div
        className="row"
      >
        <div
          className="col-2 form-col"
        >
          <input
            autoComplete="current-password"
            className="form-input"
            name="password"
            placeholder="Password"
            type="password"
          />
        </div>
        <div
          className="col-2"
        >
          <input
            autoComplete="confirm-password"
            className="form-input"
            name="confirmPassword"
            placeholder="Confirm Password"
            type="password"
          />
        </div>
      </div>
      <div
        className="row"
        id="terms-of-use"
      >
        <input
          type="checkbox"
        />
        <label>
          Agree to our
          <a
            href="#"
          >
            Terms of use
          </a>
           and
          <a
            href="#"
          >
            Privacy policy
          </a>
        </label>
      </div>
      <div>
        <input
          className="btn btn-primary form-input"
          type="submit"
          value="Sign up"
        />
      </div>
    </div>
  </form>
</div>
`;

exports[`Renders the singup page should render the signup page 1`] = `
<Fragment>
  <HeaderHome
    openModal={[Function]}
  />
  <HeroSection />
  <section>
    <Modal
      isModalOpen={false}
    >
      <div
        className="nav-tabs"
        role="tablist"
      >
        <a
          className="nav-item false"
          data-toggle="login"
          id="select-tab-login"
          onClick={[Function]}
          role="tab"
        >
          Login
        </a>
        <a
          className="nav-item false"
          data-toggle="signup"
          id="select-tab-signup"
          onClick={[Function]}
          role="tab"
        >
          Sign Up
        </a>
      </div>
      <div
        className="tab-conten row"
      >
        <LoginForm
          show={false}
        />
        <SignupForm
          show={false}
        />
      </div>
    </Modal>
  </section>
  <Footer />
</Fragment>
`;
